provider-example/examples/server.rs

```diff
@@ -0,0 +1,107 @@
+use pki_types::CertificateDer;
+use pki_types::{PrivateKeyDer, PrivatePkcs8KeyDer};
+use rustls::server::Acceptor;
+use rustls::ServerConfig;
+use rustls_provider_example::PROVIDER;
+use std::io::Write;
+use std::sync::Arc;
+
+struct TestPki {
+    server_cert_der: CertificateDer<'static>,
+    server_key_der: PrivateKeyDer<'static>,
+}
```

Comment:
As future work it might be nice to pull out the commonalities between this rcgen setup code and the code in `server_acceptor.rs`. They're quite similar.

Reply:
Indeed, ditto `simpleclient.rs`.

Reply:
Though when I original did that duplication, I couldn't find a way to make it fly without making the most simple example look odd or magical.

Reply:
Feels fair to handle separately on a rainy day.. I'll resolve this feedback.